### PR TITLE
Add package dependencies and generate without including vendor

### PR DESCRIPTION
**Pull request by andrewstucki**
This PR adds go's `dep` into the mix for vendor dependencies and adjusts the Makefile accordingly.

Developing on a Mac is simply `brew install rocksdb` followed by `make prepare` and `make`.

### CODE REVIEW

**Comment by mperham** [comment]
By any chance, did you verify the vagrant box setup and build still work?

**Comment by andrewstucki** [comment]
I can do that real quick, but they should still work given the fact that `dep` is only responsible for the building of dependent go packages and the export `CGO_CFLAGS`/`CGO_LDFLAGS` should still be passed down to the invocation of `go test`.

**Comment by mperham** [comment]
I dig it, locking down the deps was one of the things on my todo list.  Can we lock down the gorocksdb version to a commit SHA?

File: Gopkg.lock [hunk 1]
```diff
@@ -2,7 +2,6 @@
 
 
 [[projects]]
-  branch = "master"
   name = "github.com/contribsys/gorocksdb"
   packages = ["."]
   revision = "65923dcb8c3df7681d26491e7ae40abf7f3b53c5"
```
Comment (mperham) [inline Gopkg.lock]: Looks like it was locked down already? 🤷‍♂️

**Comment by mperham** [comment]
Yeah, it broke Vagrant setup but I'll fix it.

```
=== Installing dependencies
/faktory is not within a known GOPATH/src
make: *** [prepare] Error 1
```

**Pull request merged**